src/net_processing.cpp

```diff
@@ -301,6 +373,9 @@ struct CNodeState {
     }
 };
 
+// Keeps track of the time (in microseconds) when transactions were requested last time
+limitedmap<uint256, int64_t> g_already_asked_for GUARDED_BY(cs_main)(MAX_INV_SZ);
```

Comment:
Presumably the `MAX_PEER_TX_ANNOUNCEMENTS` limitation prevents any peer from triggering unnecessary evictions by overrunning this map with junk announcements.

Reply:
I think the limit on the number of in-flight requests to a peer is what prevents this from being taken over.

Reply:
Are you suggesting not limiting this map and rely only on MAX_PEER_TX_ANNOUNCEMENTS? I believe then this map would be exposed to spamming by sybils.

Reply:
Right, and you need as many as 500 sybil connections to overrun this map :)

Reply:
I forgot to add that the downside to this map being overrun is that we'll just request a transaction sooner than we otherwise would have (in the event that it was removed due to this data structure filling up). So if a zillion peers manage to take over this data structure, they're just wasting our bandwidth a bit -- which they can already do.

Reply:
> I think the limit on the number of in-flight requests to a peer is what prevents this from being taken over.

Ah right, per the third clause in this while condition: https://github.com/bitcoin/bitcoin/pull/14897/files#diff-eff7adeaec73a769788bb78858815c91R3864